### PR TITLE
Document "protected" pub/sub topics.

### PR DESCRIPTION
**Pull request by kovalromank**
## Description

Added documentation for pub/sub protected topics.  Implementation PR: dapr/dapr#6714.

## Issue reference

References dapr/dapr#5843.

### CODE REVIEW

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-scopes.md [hunk 1]
```diff
@@ -152,6 +155,50 @@ The table below shows which application is allowed to subscribe to the topics:
 | app2 | X |   |   |
 | app3 | X | X |   |
 
+## Example 4: Mark topics as protected
+
+If your topic involves sensitive data, each new application must be explicitly listed in the `publishingScopes` and `subscriptionScopes` to ensure it cannot read from or write to that topic. Alternatively, you can designate the topic as 'protected' and grant access only to specific applications that genuinely require it.
```
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-scopes.md]:
```suggestion
If your topic involves sensitive data, each new application must be explicitly listed in the `publishingScopes` and `subscriptionScopes` to ensure it cannot read from or write to that topic. Alternatively, you can designate the topic as 'protected' (using `protectedTopics`) and grant access only to specific applications that genuinely require it.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-scopes.md [hunk 2]
```diff
@@ -152,6 +155,50 @@ The table below shows which application is allowed to subscribe to the topics:
...
+    value: ""
+  - name: protectedTopics
+    value: "A,B"
+  - name: publishingScopes
+    value: "app1=A,B;app2=B"
+  - name: subscriptionScopes
+    value: "app1=A,B;app2=B"
+```
+
+> Note that topics A and B are marked as protected. As a result, even though app3 is not listed under `publishingScopes` or `subscriptionScopes` it cannot interact with these topics.
```
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-scopes.md]:
```suggestion
In the example above, topics A and B are marked as protected. As a result, even though `app3` is not listed under `publishingScopes` or `subscriptionScopes`, it cannot interact with these topics.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
just a quick review - thanks!

**Review by hhunter-ms — APPROVED** [review]

**Review by msfussell — APPROVED** [review]
LGTM. I have raised this issue https://github.com/dapr/docs/issues/3657 to improve the referencing to this topic.

**Pull request merged**